api/v1alpha1/nifidataflow_types.go

```diff
@@ -151,11 +153,14 @@ func (d *NifiDataflowSpec) SyncNever() bool {
 	return d.GetSyncMode() == SyncNever
 }
 
-func (d *NifiDataflowSpec) GetParentProcessGroupID(rootProcessGroupId string) string {
-	if d.ParentProcessGroupID == "" {
+func (d *NifiDataflowSpec) GetParentProcessGroupID(rootProcessGroupId string, parentProcessGroupRef *NifiResource) string {
+	if parentProcessGroupRef != nil && parentProcessGroupRef.Status.UUID != "" {
+		return parentProcessGroupRef.Status.UUID
+	} else if d.ParentProcessGroupID == "" {
 		return rootProcessGroupId
+	} else {
+		return d.ParentProcessGroupID
```

Comment:
Should be removed.

`NifiDataflow` is in `v1`, the `v1alpha1`must not be updated.